### ISSUE TITLE
/usr/bin/ld: cannot find -lpthreads

### ISSUE THREAD

**Issue by jean-christophe-manciot** [opensourcerouting/libyang-debian#1]
Ubuntu disco
CESNET libyang v0.16-r3 (0.16.105)
libc6-dev 2.28-4

Building libyang with libyang-debian with:
```
debuild -i -I --no-sign --build=binary -j1
```
leads to:
```
...
Run Build Command:"/usr/bin/make" "cmTC_9c271/fast"
make[2]: Entering directory '/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
/usr/bin/make -f CMakeFiles/cmTC_9c271.dir/build.make CMakeFiles/cmTC_9c271.dir/build
make[3]: Entering directory '/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_9c271.dir/CheckFunctionExists.c.o
/usr/bin/cc   -g -O2 -fdebug-prefix-map=/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -DCHECK_FUNCTION_EXISTS=pthread_create -fPIE   -o CMakeFiles/cmTC_9c271.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.13/Modules/CheckFunctionExists.c
Linking C executable cmTC_9c271
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_9c271.dir/link.txt --verbose=1
/usr/bin/cc -g -O2 -fdebug-prefix-map=/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -DCHECK_FUNCTION_EXISTS=pthread_create  -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now  -rdynamic CMakeFiles/cmTC_9c271.dir/CheckFunctionExists.c.o  -o cmTC_9c271 -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/cmTC_9c271.dir/build.make:87: cmTC_9c271] Error 1
make[3]: Leaving directory '/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
make[2]: *** [Makefile:121: cmTC_9c271/fast] Error 2
make[2]: Leaving directory '/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
```
Any suggestion?

**Comment by jean-christophe-manciot** [comment]
From the build log, I get:
```
...
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found PCRE: /usr/lib/x86_64-linux-gnu/libpcre.so  
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components:  doxygen dot 
-- Found CMocka: /usr/lib/x86_64-linux-gnu/libcmocka.so (Required is at least version "1.0.0") 
-- The CXX compiler identification is GNU 8.2.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found SWIG: /usr/bin/swig3.0 (found suitable version "3.0.12", minimum required is "3.0.12") 
-- Python version 3.7;3.7dm was selected
CMake Error at swig/CMakeLists.txt:28 (if):
  if given arguments:
    "3.7" "3.7dm" "STREQUAL" "2"
  Unknown arguments specified

-- Configuring incomplete, errors occurred!
See also "/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeOutput.log".
See also "/media/actionmystique/SAMSUNG5-Shared/home/actionmystique/src/Yang/libyang-build/git-libyang/obj-x86_64-linux-gnu/CMakeFiles/CMakeError.log".
	cd obj-x86_64-linux-gnu && tail -v -n \+0 CMakeCache.txt
```
From CMakeError.log, ```Determining if the function pthread_create exists in the pthreads failed```.

However, it is there:
```
# find /lib -name "*pthread*"
/lib/x86_64-linux-gnu/libpthread-2.28.so
/lib/x86_64-linux-gnu/libpthread.so.0
# nm /lib/x86_64-linux-gnu/libpthread.so.0 | grep "pthread_create"
0000000000008280 t __pthread_create_2_1
00000000000065be t __pthread_create_2_1.cold.12
0000000000008280 T pthread_create@@GLIBC_2.2.5
```

More details with:
- [libyang_0.16.105_amd64.build.txt](https://github.com/opensourcerouting/libyang-debian/files/2768070/libyang_0.16.105_amd64.build.txt)
- [CMakeError.log](https://github.com/opensourcerouting/libyang-debian/files/2768076/CMakeError.log)
- [CMakeOutput.log](https://github.com/opensourcerouting/libyang-debian/files/2768078/CMakeOutput.log)

**Comment by eqvinox** [comment]
Hm. Interesting. I'm looking at this but I have no idea yet what might be causing this.  I'll go install an Ubuntu VM to test — I assume there is nothing "special" you did?

**Comment by eqvinox** [comment]
Can you please try 7c99ac8a9eef03c6d38372463bc595ab4ceff570?  For the moment I can't reproduce your issue, it's building just fine on Ubuntu disco in my container :neutral_face:

**Comment by njh** [comment]
Weirdly I had a problem with missing `-lpthreads` when it was actually failing to find swig 3.0.

**Comment by eqvinox** [comment]
Closing since libyang 0.16.105-1 is now available officially in Ubuntu 19.04 (disco)
cf. https://launchpad.net/ubuntu/+source/libyang
